bottomnavigation/bottomnavigation/src/main/java/com/microsoft/device/dualscreen/bottomnavigation/BottomNavigationView.kt

```diff
@@ -16,76 +17,132 @@ import android.view.MotionEvent
-import com.microsoft.device.dualscreen.createHalfTransparentBackground
-import com.microsoft.device.dualscreen.isPortrait
-import com.microsoft.device.dualscreen.isSpannedInDualScreen
+import com.microsoft.device.dualscreen.utils.wm.DisplayPosition
+import com.microsoft.device.dualscreen.utils.wm.OnSwipeListener
+import com.microsoft.device.dualscreen.utils.wm.ScreenMode
+import com.microsoft.device.dualscreen.utils.wm.areScreensSideBySide
+import com.microsoft.device.dualscreen.utils.wm.createHalfTransparentBackground
+import com.microsoft.device.dualscreen.utils.wm.getHinge
+import com.microsoft.device.dualscreen.utils.wm.getWindowRect
+import com.microsoft.device.dualscreen.utils.wm.isSpanned
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.mapNotNull
+import kotlinx.coroutines.launch
 
 /**
  * A sub class of the Bottom Navigation View that can position its children in different ways when the application is spanned on both screens.
  * Using the [arrangeButtons] and [displayPosition] the children can be split in any way between the two screens.
  * Animation can be used when changing the arrangement of the buttons on the two screen.
  * If one of the screens doesn't contain any button, its background can be made transparent.
  */
-open class SurfaceDuoBottomNavigationView @JvmOverloads constructor(
-    context: Context,
-    attrs: AttributeSet? = null,
-    defStyleAttr: Int = 0
-) : BottomNavigationView(context, attrs, defStyleAttr) {
+open class BottomNavigationView : BottomNavigationView {
+    constructor(context: Context) : super(context) {
+        initFoldingObs()
+    }
+
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
+        initFoldingObs()
+        extractAttributes(context, attrs)
+    }
 
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
+        context,
+        attrs,
+        defStyleAttr
+    ) {
+        initFoldingObs()
+        extractAttributes(context, attrs)
+    }
+
+    private var job: Job? = null
     private var singleScreenWidth = -1
     private var totalScreenWidth = -1
     private var hingeWidth = -1
 
     private var screenMode = ScreenMode.DUAL_SCREEN
-    private var currentScreenInfo: ScreenInfo? = null
+    private var foldingFeature: FoldingFeature? = null
 
     private var initialBackground: Drawable? = null
     private var startBtnCount: Int = -1
     private var endBtnCount: Int = -1
     private var defaultChildWidth = -1
 
-    private var onSwipeListener: OnSwipeListener = object : OnSwipeListener(context) {
-        override fun onSwipeLeft() {
-            super.onSwipeLeft()
-            if (allowFlingGesture) {
-                displayPosition = DisplayPosition.START
-            }
+    private fun initFoldingObs() {
```

Comment:
What does the functions' name mean? To me is not descriptive enough (mainly the `Obs` part).